HelpSource/Classes/History.schelp

```diff
@@ -56,44 +133,98 @@ store in a file, in historical order as individual code snippets.
 method::loadStory
 read history into current, from a file in story format.
 
+method::rewrite
+Write a properly formatted code file from a history.
+argument::path
+The filename is the original name with "_rewritten." appended.
+argument::open
+If open is true (default: true), open a text window with the string.
+
+method::readFromDoc
+read in a history from a code file created with .rewrite
+
+
+subsection:: editing history
+method::clear
+remove all items from (current) history.
+
+method::enter
+add an entry to (current) history by hand.
+
+method::drop
+drop the newest n lines from history. if n is negative, drop the oldest n lines.
+
+method::keep
+keep only the newest n lines from history. if n is negative, keep the oldest n lines.
+
+method::removeAt
+remove a specific line
+
+method::removeLast
+remove last line from history
+
+subsection:: repeating history
+method::evaluateLineAt
+
 method::play
 play back current history from start to end line, per default verbose.
 
 method::stop
 stop current history playback.
 
-method::rewrite
-Write a properly formatted code file from a history.
 
-argument::path
-The filename is the original name with "_rewritten." appended.
+subsection:: date, time, string functions
+method::startTimeStamp
+method::formatTime
+method::unformatTime
+method::shorten
+method::prettyString
 
-argument::open
-If open is true (default: true), open a text window with the string.
+private:: checkPath
 
-subsection::More internal methods:
+method::new
+create a new instance.
 
-method:: current
-the current History instance
 
-method:: lines, lineShorts
-the currently recorded lines in History.current.
+InstanceMethods::
 
-lineShorts is a copy with shortened strings for display.
+Note:: the instance methods code::[hasMovedOn_, hasMovedOn, play, stop, lines, lineShorts, removeAt, removeLast, keep, drop, clear, saveCS, loadCS, saveStory, evaluateLineAt, loadStory, makeWin, document]:: are also implemented as class methods, and documented above.
+::
 
-method:: new
-create a new instance.
+private:: hasMovedOn_, hasMovedOn, play, stop, lines, lineShorts, removeAt, removeLast, keep, drop, clear, saveCS, loadCS, saveStory, evaluateLineAt, loadStory, makeWin, document
+
+method::isCurrent
+flag whether this history is History.current
+
+method::makeCurrent
+make this history History.current
+
+method::player
+the player task for ths history
+
+method::keys
+all keys of code authors in this history
+
+method::addLine
+add a line to history with current time, id, and code string
+
+
+method::indicesFor
+find line indices created by keys and containing string. used for filtering.
+
+private:: matchKeys, matchString, storyString
 
 Examples::
 
 code::
+// same as first example:
```

Comment:
in these code examples, the whitespace between the code and end-of-line comments is filled with tabs and as a result gets misaligned. while you're working on this file, that might be a good thing to fix.

Reply:
done.